soepy/python/shared/shared_auxiliary.py

```diff
@@ -78,8 +80,19 @@ def calculate_log_wage_systematic(model_params, states, covariates):
     return log_wage_systematic
 
 
-def calculate_nonconsumption_utility(model_params):
+def calculate_non_consumption_utility(model_params, states):
     """Calculate non-pecuniary utility contribution."""
-    nonconsumption_utility = np.exp([0, model_params.theta_p, model_params.theta_f])
 
-    return nonconsumption_utility
+    non_consumption_utility = np.full(
+        (states.shape[0], NUM_CHOICES),
+        np.exp([0, model_params.const_p, model_params.const_f]),
+    )
+
+    for i in range(1, model_params.num_types):
+        non_consumption_utility[np.where(states[:, 5] == i)] += [
```

Comment:
IMPORTANT: Doesn't this need to be a multiplication for the shifts? Please carefully revisit.

Reply:
Addressed

Reply:
Addressed